### ISSUE TITLE
[Bug]: OOM issue when doing `bal build` to ballerina client code

### ISSUE THREAD

**Issue by lnash94** [ballerina-platform/ballerina-lang#41205]
### Description

I tried `bal build` with this ballerina code[1], then found this OOM issue intermittently. Here I have attached the ballerina source code[1] and hprof files[2]

[1] [avatax.zip](https://github.com/ballerina-platform/ballerina-lang/files/12332943/avatax.zip)
[2] Here is the link for hprof files: https://drive.google.com/drive/folders/1B-vMz6JnT5ChV9WDDjrwBubf00V_eROt?usp=sharing
Ballerina version : 2201.7.0
```
....
We thank you for helping make us better.

[2023-08-14 13:49:25,831] SEVERE {b7a.log.crash} - Java heap space 
java.lang.OutOfMemoryError: Java heap space
        at java.base/java.lang.StringUTF16.newBytesFor(StringUTF16.java:52)
        at java.base/java.lang.AbstractStringBuilder.inflate(AbstractStringBuilder.java:228)
        at java.base/java.lang.AbstractStringBuilder.putStringAt(AbstractStringBuilder.java:1665)
        at java.base/java.lang.AbstractStringBuilder.append(AbstractStringBuilder.java:539)
        at java.base/java.lang.StringBuilder.append(StringBuilder.java:178)
        at org.ballerinalang.observability.anaylze.DefaultObservabilitySymbolCollector.generateSyntaxTreeJsonString(DefaultObservabilitySymbolCollector.java:213)
        at org.ballerinalang.observability.anaylze.DefaultObservabilitySymbolCollector.writeToExecutable(DefaultObservabilitySymbolCollector.java:126)
        at io.ballerina.projects.JarResolver.getJarFilePathsRequiredForExecution(JarResolver.java:129)
        at io.ballerina.projects.JBallerinaBackend.emitExecutable(JBallerinaBackend.java:533)
        at io.ballerina.projects.JBallerinaBackend.emit(JBallerinaBackend.java:215)
        at io.ballerina.cli.task.CreateExecutableTask.execute(CreateExecutableTask.java:114)
        at io.ballerina.cli.TaskExecutor.executeTasks(TaskExecutor.java:40)
        at io.ballerina.cli.cmd.BuildCommand.execute(BuildCommand.java:288)
        at io.ballerina.cli.launcher.Main$$Lambda$31/0x0000000800103840.accept(Unknown Source)
        at java.base/java.util.Optional.ifPresent(Optional.java:183)
        at io.ballerina.cli.launcher.Main.main(Main.java:56)
```
Atteched : hprof analys
<img width="1512" alt="Screenshot 2023-08-14 at 13 54 01" src="https://github.com/ballerina-platform/ballerina-lang/assets/25495666/47affe60-6b9b-4e7c-aedf-af1f5eaa7f87">

### Steps to Reproduce

Try `bal build` for this ballerina source code [1] [avatax.zip](https://github.com/ballerina-platform/ballerina-lang/files/12332943/avatax.zip)

### Affected Version(s)

2201.7.0

### Related area

-> Compilation

**Comment by gabilang** [comment]
The fix added with the PR https://github.com/ballerina-platform/ballerina-lang/pull/41348 will resolve this issue.

**Comment by NipunaMadhushan** [comment]
Closing this issue since it has been fixed with #41348
